.circleci/config.yml

```diff
@@ -264,136 +402,98 @@ jobs:
-  puppeteer-e2e-test:
-    <<: *defaults
-    docker:
-      - image: circleci/node:13.10.1-browsers
-        environment:
-          NODE_ENV: development
-    working_directory: ~/workbench
-    parallelism: 3
-    steps:
-      - checkout
-      - run: *update_git_submodules
-      - restore_cache:
-          keys:
-          - v2-e2e-test-cache-{{ checksum "~/workbench/e2e/yarn.lock" }}
-          - v2-e2e-test-cache-
-      - run:
-          name: Initialize Yarn
-          working_directory: ~/workbench/e2e
-          command: yarn cache clean && yarn install --frozen-lockfile
-      - save_cache:
-          paths:
-            - ~/.cache/yarn
-          key: v2-e2e-test-cache-{{ checksum "~/workbench/e2e/yarn.lock" }}
-      - run:
-          name: Update Puppeteer Environment Variables
-          command: |
-            echo 'export USER_NAME=$PUPPETEER_TEST_USER' >> $BASH_ENV
-            echo 'export PASSWORD=$PUPPETEER_TEST_USER_PASSWORD' >> $BASH_ENV
-            echo 'export INVITATION_KEY=$PUPPETEER_TEST_REGISTRATION_KEY' >> $BASH_ENV
-            source $BASH_ENV
-      - run:
-          name: Run e2e tests (Puppeteer and Jest)
-          working_directory: ~/workbench/e2e
-          command: yarn test:ci $(circleci tests glob "tests/**/*.spec.ts" | circleci tests split --split-by=timings --timings-type=classname --show-counts)
-      - store_artifacts:
-          path: e2e/logs
-          destination: logs
-      - store_test_results:
-          path: ~/workbench/e2e/logs
       
   ui-deploy-to-test:
-    <<: *defaults
+    executor: workbench_docker
     steps:
-      - checkout
-      - run: *update_git_submodules
-      # Use the ui-build-test workspace here to avoid redoing the setup.
+      # - git_checkout
+      # Use "ui-build-test" job's workspace here to avoid redoing the setup.
       - attach_workspace:
           at: .
-      - restore_cache:
-          keys:
-          - ui-cache-{{ checksum "~/workbench/ui/yarn.lock" }}
-          - ui-cache-
-      - deploy:
-          name: Deploy UI App to App Engine
-          working_directory: ~/workbench/ui
-          command: |
-            ../ci/activate_creds.sh circle-sa-key.json
-            ./project.rb deploy-ui \
-              --project all-of-us-workbench-test \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
-              --version circle-ci-test \
-              --key-file circle-sa-key.json \
-              --promote
+      - ui_dependencies
+      - activate_service_accont_creds
+      - deploy_appengine:
+          project_name: "all-of-us-workbench-test"
+          deploy_arg: "deploy-ui"
+          working_dir: ~/workbench/ui
+
 
   # In staging and perf, we deploy all at once in order to track and control
   # the release for a tagged version only. In "test", we maintain individual
   # release jobs for API and UI to allow faster deploys and because we lack
   # a version tag.
   deploy-staging:
```

Comment:
Can we make the `deploy_` jobs parameterized by the destination environment (maybe later)?